src/components/__tests__/Subthoughts.js

```diff
@@ -168,3 +169,48 @@ describe('context view', () => {
   })
 
 })
+
+it('generate proper thoughtsResolved for circular path in context view', async () => {
+
+  // import thoughts
+  await store.dispatch(importText(RANKED_ROOT, `
+  - a
+    - b
+      - c
+        - d
+          - c`))
+
+  store.dispatch({ type: 'setCursor', thoughtsRanked: [
+    { value: 'a', rank: 0 },
+    { value: 'b', rank: 0 },
+    { value: 'c', rank: 0 },
+    { value: 'd', rank: 0 },
+    { value: 'c', rank: 0 },
+  ]
+  })
+
+  store.dispatch({ type: 'toggleContextView' })
```

Comment:
We have middleware that supports passing an array of actions to `store.dispatch`.